webui/react/src/hooks/useModal/HyperparameterSearch/useModalHyperparameterSearch.tsx

```diff
@@ -359,7 +359,7 @@ const useModalHyperparameterSearch = ({
         Object.values(SEARCH_METHODS).find((searcher) => searcher.name === searcherName) ??
         SEARCH_METHODS.ASHA;
       setSearcher(searcher);
-      form.setFieldValue('searcher', searcher);
+      form.setFieldValue('searcher', searcher.name);
```

Comment:
the previous implementation returned all object https://github.com/determined-ai/determined/blob/c010993fa6571edd57d6e19009add3b14df6c540/webui/react/src/hooks/useModal/HyperparameterSearch/useModalHyperparameterSearch.tsx#L65-L81

However, this code https://github.com/determined-ai/determined/blob/c010993fa6571edd57d6e19009add3b14df6c540/webui/react/src/hooks/useModal/HyperparameterSearch/useModalHyperparameterSearch.tsx#L141 expects only 'single', 'random', 'grid', 'custom', or 'adaptive_asha' since i got this error 
<img width="747" alt="Screenshot 2023-06-29 at 2 19 38 PM" src="https://github.com/determined-ai/determined/assets/109113616/c6c0b338-a8f1-4d74-ba20-d6b152595fb9">.
That means `searcher.name` type should be string, so i set field value `searcher.name` instead of `searcher`. This also fixes the yaml parsing.